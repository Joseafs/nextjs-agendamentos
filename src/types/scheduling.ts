export type TpSchedulingItem = {
  title: string;
  timeStart: string;
  timeEnd: string;
};
