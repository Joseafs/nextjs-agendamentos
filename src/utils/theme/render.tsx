const isProd = process.env.NODE_ENV === 'production';

const rootPath = isProd ? process.env.NEXT_PUBLIC_DOMAIN : '/';

export const fixPath = (path = ''): string => `${rootPath}${path}`;
