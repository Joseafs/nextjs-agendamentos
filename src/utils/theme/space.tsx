import { useTheme } from 'styled-components';

export const spaceCalc = (space: number[], type: 'margin' | 'padding') => {
  const theme = useTheme();

  const calc = (value: number) => value * theme.space;

  return space.length === 1
    ? `${type}: ${calc(space[0])}px`
    : space.length === 2
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px`
    : space.length === 3
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px ${calc(space[2])}px`
    : space.length === 4
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px ${calc(
        space[2]
      )}px ${calc(space[3])}px`
    : ``;
};
