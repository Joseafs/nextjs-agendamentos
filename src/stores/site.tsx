import React, { createContext, useEffect, useState } from 'react';
import { TpErrorList, TpSchedulingItem } from '~/types/common';

const useSiteContext = () => {
  const [scheduling, setScheduling] = useState<TpSchedulingItem[]>([]);
  const [schedulingConflicts, setSchedulingConflicts] = useState<
    TpSchedulingItem[]
  >([]);
  const [error, setError] = useState<TpErrorList[]>([]);

  const remErrorByName = (name: string) => {
    const newErrorList = error.filter((item) => item.id !== name);
    setError(newErrorList);
  };
  const addErrorByName = (name: string, newText: string | JSX.Element) => {
    const newErrorList = error.filter((item) => item.id !== name);

    const newError = {
      id: name,
      text: newText
    };
    setError([...newErrorList, newError]);
  };
  const remErrorAll = () => setError([]);

  useEffect(() => {
    if (scheduling.length < 1) {
      setError([]);
    }
  }, [scheduling]);

  return {
    state: {
      scheduling,
      schedulingConflicts,
      error
    },
    dispatch: {
      setScheduling,
      setSchedulingConflicts,
      setError,
      setErrorRemByName: remErrorByName,
      setErrorAddByName: addErrorByName,
      setErrorRemAll: remErrorAll
    }
  };
};

type PropsContextBase = ReturnType<typeof useSiteContext>;

export const SiteContext = createContext({} as PropsContextBase);

interface PropsSiteStore {
  children: React.ReactNode;
}

export const SiteStore = ({ children }: PropsSiteStore) => {
  return (
    <SiteContext.Provider value={useSiteContext()}>
      {children}
    </SiteContext.Provider>
  );
};
