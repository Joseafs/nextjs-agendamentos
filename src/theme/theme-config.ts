// import PoppinsTtf from '../fonts/poppins/Poppins-Regular.ttf';
import { css } from 'styled-components';
import { PropsTheme } from '~/types/theme';

export const theme: PropsTheme = {
  name: 'default',
  breakpoints: {
    keys: ['xs', 'sm', 'md', 'lg', 'xl'],
    width: {
      desktop: '1120px',
      tablet: '740px',
      mobile: '360px'
    },
    values: {
      xs: 512,
      sm: 768,
      md: 1024,
      lg: 1280,
      xl: 1536
    }
  },
  palette: {
    mode: 'light',
    common: {
      black: '#000',
      white: '#fff'
    },
    primary: {
      main: '#0033CC',
      light: '#3DA8F5',
      dark: '#002699',
      text: '#fff'
    },
    secondary: {
      main: '#0FAC0F'
    },
    error: {
      main: '#C20A0A',
      light: '#F76D6D',
      dark: '#AA0909',
      text: '#FEE7E7'
    },
    warning: {
      main: '#FEF6E7'
    },
    info: {
      main: '#E5ECFF'
    },
    success: {
      main: '#08916F'
    },
    grey: {
      50: '#fafafa',
      100: '#f5f5f5',
      200: '#eeeeee',
      300: '#e0e0e0',
      400: '#bdbdbd',
      500: '#9e9e9e',
      600: '#757575',
      700: '#666666',
      800: '#495057',
      900: '#333333',
      1000: '#111'
    },
    text: {
      primary: 'rgba(0,0,0,.87)',
      secondary: 'rgba(0,0,0,.6)',
      disabled: 'rgba(0,0,0,.38)'
    },
    background: {
      default: '#fff',
      divider: 'rgba(0,0,0,.1)',
      rgba: 'rgba(255,255,255,.7)'
    }
  },
  shape: {
    radius: 4
  },
  space: 8,
  shadows: [
    'none',
    '0 4px 8px rgba(0, 0, 0, 0.1)',
    '0 -4px 8px 0 rgba(0, 0, 0, 0.08)',
    '0 0 6px 0px rgba(0, 0, 0, .4)',
    '0 6px 8px rgba(0, 0, 0, 0.2)'
  ],
  zindex: {
    screen: 1000,
    appBar: 1100,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
    menu: 2000
  },
  animation: {
    transition: [
      'transition: all 0.2s ease-in-out',
      'transition: all 0.4s ease-in-out'
    ],
    hover: [
      css`
        background-color: ${({ theme }) => theme.palette.opacity.light};
      `,
      css`
        background-color: rgba(0, 0, 0, 0.2);
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      `,
      css`
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      `,
      css`
        background-color: rgba(0, 0, 0, 0.1);
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.1);
        transform: translateY(-4px);
      `
    ],
    active: [
      'transform: scale(0.94); transition: unset;',
      'transform: scale(0.99); transition: unset;'
    ]
  }
};

export default theme;
