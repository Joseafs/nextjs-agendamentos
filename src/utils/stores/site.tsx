import React, { createContext, useState } from 'react';
import { TpSchedulingItem } from '~/types/scheduling';

const useValue = () => {
  const [scheduling, setScheduling] = useState<TpSchedulingItem[]>([]);

  return {
    state: {
      scheduling
    },
    dispatch: {
      setScheduling
    }
  };
};

export const SiteContext = createContext({} as ReturnType<typeof useValue>);

interface PropsSiteStore {
  children: React.ReactNode;
}

export const SiteStore = (props: PropsSiteStore) => {
  return (
    <SiteContext.Provider value={useValue()}>
      {props.children}
    </SiteContext.Provider>
  );
};
