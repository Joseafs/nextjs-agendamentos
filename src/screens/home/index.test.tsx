import { act, fireEvent, render } from '@testing-library/react';
import { SiteContext } from '~/stores/site';
import { ThemeUI } from '~/theme/theme-provider';
import { TpSchedulingItem } from '~/types/common';
import { ScreenHome } from '.';

const mockSetError = jest.fn();
const mockSetErrorAddByName = jest.fn();
const mockSetErrorRemByName = jest.fn();
const mockSetErrorRemAll = jest.fn();
const mockSetScheduling = jest.fn();
const mockSetSchedulingConflicts = jest.fn();

const listScheduling = [
  {
    id: 1,
    title: 'Test Note A',
    dateTimeStart: '2023-01-03T15:22',
    dateTimeEnd: '2023-01-03T22:24'
  },

  {
    id: 2,
    title: 'Test Note B',
    dateTimeStart: '2000-02-03T01:22',
    dateTimeEnd: '2000-02-03T02:24'
  },
  {
    id: 3,
    title: 'Test Note C',
    dateTimeStart: '2021-01-03T01:22',
    dateTimeEnd: '2021-01-03T02:24'
  }
];

const componentScheduling = (conflicts: TpSchedulingItem[]) => (
  <SiteContext.Provider
    value={{
      state: {
        scheduling: listScheduling,
        schedulingConflicts: conflicts,
        error: []
      },
      dispatch: {
        setError: mockSetError,
        setErrorAddByName: mockSetErrorAddByName,
        setErrorRemByName: mockSetErrorRemByName,
        setErrorRemAll: mockSetErrorRemAll,
        setScheduling: mockSetScheduling,
        setSchedulingConflicts: mockSetSchedulingConflicts
      }
    }}
  >
    <ThemeUI>
      <ScreenHome />
    </ThemeUI>
  </SiteContext.Provider>
);

describe('ScreenHome', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('Should match snapshot', () => {
    const { container } = render(componentScheduling([]));
    expect(container).toMatchSnapshot();
  });

  it('Should call setScheduling when click to remove', async () => {
    const { getByTestId } = render(componentScheduling([listScheduling[0]]));
    const btnRemove = getByTestId('button-icon--remove-1') as HTMLButtonElement;

    await act(async () => {
      fireEvent.click(btnRemove);
    });

    expect(mockSetScheduling).toHaveBeenCalledTimes(1);
    expect(mockSetScheduling).toBeCalledWith([
      listScheduling[1],
      listScheduling[2]
    ]);
  });
  it('Should call setScheduling when remove all rows', async () => {
    const { getByTestId } = render(componentScheduling([listScheduling[0]]));
    const btnRemove1 = getByTestId(
      'button-icon--remove-1'
    ) as HTMLButtonElement;
    const btnRemove2 = getByTestId(
      'button-icon--remove-2'
    ) as HTMLButtonElement;
    const btnRemove3 = getByTestId(
      'button-icon--remove-3'
    ) as HTMLButtonElement;

    await act(async () => {
      fireEvent.click(btnRemove1);
      fireEvent.click(btnRemove2);
      fireEvent.click(btnRemove3);
    });

    expect(mockSetScheduling).toBeCalledTimes(3);
  });
});
