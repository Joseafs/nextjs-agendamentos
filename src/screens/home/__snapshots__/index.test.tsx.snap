// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHome Should match snapshot 1`] = `
<div>
  <div
    class="sc-e0c7811d-0 bFbHQj"
  >
    <main
      class="sc-e0c7811d-1 hHeTCp"
    >
      <section
        class="sc-acbe88d-1 ezYntl"
      >
        <div
          class="sc-acbe88d-0 hSHwjd"
        >
          <h1
            class="sc-9f5fb0c5-0 cpBmvw"
          >
            Agendamento Online
          </h1>
          <form
            action="#"
          >
            <div
              class="sc-4a45138a-0 cEXHx"
              id="formik--scheduling-root"
            >
              <div
                class="sc-8c4bf41-0 kZOgiB"
              >
                <div
                  class="sc-277720d7-0 bneybE"
                >
                  <label
                    class="sc-277720d7-1"
                    for="title"
                  >
                    Título
                  </label>
                  <input
                    class="sc-277720d7-2 gznZqK"
                    data-testid="input-title"
                    id="title"
                    name="title"
                    placeholder="Informe o Título do agendamento"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="sc-4a45138a-1 fyXKeg"
              >
                <div
                  class="sc-8c4bf41-0 kZOgiB"
                >
                  <div
                    class="sc-277720d7-0 bneybE"
                  >
                    <label
                      class="sc-277720d7-1"
                      for="dateTimeStart"
                    >
                      Dia e horário de início
                    </label>
                    <input
                      class="sc-277720d7-2 gznZqK"
                      data-testid="input-dateTimeStart"
                      id="dateTimeStart"
                      max=""
                      name="dateTimeStart"
                      required=""
                      type="datetime-local"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="sc-8c4bf41-0 kZOgiB"
                >
                  <div
                    class="sc-277720d7-0 bneybE"
                  >
                    <label
                      class="sc-277720d7-1"
                      for="dateTimeEnd"
                    >
                      Dia e horário de fim
                    </label>
                    <input
                      class="sc-277720d7-2 gznZqK"
                      data-testid="input-dateTimeEnd"
                      id="dateTimeEnd"
                      min=""
                      name="dateTimeEnd"
                      required=""
                      type="datetime-local"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="sc-8c4bf41-0 hnGkhO"
              >
                <button
                  class="sc-1b27c299-0 ivtgTd"
                  color="primary"
                  disabled=""
                  type="submit"
                >
                  Agendar
                </button>
              </div>
            </div>
          </form>
          <div
            class="sc-36d0b95e-0 OBoBO"
          >
            <table
              class="sc-36d0b95e-3 cQqYlF"
            >
              <thead>
                <tr>
                  <th
                    class="sc-36d0b95e-1 evtReD"
                  >
                    Título
                    <div
                      class="sc-36d0b95e-4 ctEyvB"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 1a.5.5 0 0 1 .5.5v11.793l3.146-3.147a.5.5 0 0 1 .708.708l-4 4a.5.5 0 0 1-.708 0l-4-4a.5.5 0 0 1 .708-.708L7.5 13.293V1.5A.5.5 0 0 1 8 1z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </th>
                  <th
                    class="sc-36d0b95e-1 evtReD"
                  >
                    Início
                    <div
                      class="sc-36d0b95e-4 fFbXZW"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 1a.5.5 0 0 1 .5.5v11.793l3.146-3.147a.5.5 0 0 1 .708.708l-4 4a.5.5 0 0 1-.708 0l-4-4a.5.5 0 0 1 .708-.708L7.5 13.293V1.5A.5.5 0 0 1 8 1z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </th>
                  <th
                    class="sc-36d0b95e-1 evtReD"
                  >
                    Fim
                    <div
                      class="sc-36d0b95e-4 ctEyvB"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 1a.5.5 0 0 1 .5.5v11.793l3.146-3.147a.5.5 0 0 1 .708.708l-4 4a.5.5 0 0 1-.708 0l-4-4a.5.5 0 0 1 .708-.708L7.5 13.293V1.5A.5.5 0 0 1 8 1z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </th>
                  <th
                    align="center"
                  >
                    Duração
                  </th>
                  <th
                    align="center"
                  >
                    Apagar
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class=""
                >
                  <td
                    align="center"
                    data-th="Título"
                  >
                    Test Note A
                  </td>
                  <td
                    align="center"
                    data-th="Início"
                  >
                    03/01/2023
                     
                    <br />
                     
                    15:22
                  </td>
                  <td
                    align="center"
                    data-th="Fim"
                  >
                    03/01/2023
                     
                    <br />
                     
                    22:24
                  </td>
                  <td
                    align="center"
                    data-th="Duração"
                  >
                    7 horas e 2 minutos
                  </td>
                  <td
                    align="center"
                    width="120"
                  >
                    <button
                      class="sc-f162131e-0 jLjXKu sc-36d0b95e-2 feocFU"
                      color="error"
                      data-testid="button-icon--remove-1"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1.2rem"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1.2rem"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class=""
                >
                  <td
                    align="center"
                    data-th="Título"
                  >
                    Test Note C
                  </td>
                  <td
                    align="center"
                    data-th="Início"
                  >
                    03/01/2021
                     
                    <br />
                     
                    01:22
                  </td>
                  <td
                    align="center"
                    data-th="Fim"
                  >
                    03/01/2021
                     
                    <br />
                     
                    02:24
                  </td>
                  <td
                    align="center"
                    data-th="Duração"
                  >
                    1 hora e 2 minutos
                  </td>
                  <td
                    align="center"
                    width="120"
                  >
                    <button
                      class="sc-f162131e-0 jLjXKu sc-36d0b95e-2 feocFU"
                      color="error"
                      data-testid="button-icon--remove-3"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1.2rem"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1.2rem"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class=""
                >
                  <td
                    align="center"
                    data-th="Título"
                  >
                    Test Note B
                  </td>
                  <td
                    align="center"
                    data-th="Início"
                  >
                    03/02/2000
                     
                    <br />
                     
                    01:22
                  </td>
                  <td
                    align="center"
                    data-th="Fim"
                  >
                    03/02/2000
                     
                    <br />
                     
                    02:24
                  </td>
                  <td
                    align="center"
                    data-th="Duração"
                  >
                    1 hora e 2 minutos
                  </td>
                  <td
                    align="center"
                    width="120"
                  >
                    <button
                      class="sc-f162131e-0 jLjXKu sc-36d0b95e-2 feocFU"
                      color="error"
                      data-testid="button-icon--remove-2"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1.2rem"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1.2rem"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;
